spec/rails_helper.rb

```diff
@@ -1,15 +1,19 @@
 require 'spec_helper'
 ENV['RAILS_ENV'] ||= 'test'
 require File.expand_path('../../config/environment', __FILE__)
+require 'support/factory_bot'
 
 abort('The Rails environment is running in production mode!') if Rails.env.production?
 require 'rspec/rails'
+require_relative 'support/controller_macros'
 
 require 'capybara/rspec'
 ActiveRecord::Migration.maintain_test_schema!
 
 RSpec.configure do |config|
   config.fixture_path = "#{::Rails.root}/spec/fixtures"
+  config.include Devise::Test::ControllerHelpers, type: :controller
+  config.extend ControllerMacros, type: :controller
```

Comment:
@sinpai let's move those including and extending at the end of configure block. It's external logic, so shouldn't go with standard config setup

Reply:
👍